src/combinatorics/set_family/intersecting.lean

```diff
@@ -0,0 +1,191 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.fintype.basic
+import order.upper_lower
+
+/-!
+# Intersecting families
+
+This file defines intersecting families and proves their basic properties.
+
+## Main declarations
+
+* `set.intersecting`: Predicate for a set of elements in a generalized boolean algebra to be an
+  intersecting family.
+* `set.intersecting.card_le`: An intersecting family can only take up to half the elements, because
+  `a` and `aᶜ` cannot simultaneously be in it.
+* `set.intersecting.is_max_iff_card_eq`: Any maximal intersecting family takes up half the elements.
+
+## References
+
+* [D. J. Kleitman, *Families of non-disjoint subsets*][kleitman1966]
+-/
+
+open finset
+
+variables {α : Type*}
+
+namespace set
+section generalized_boolean_algebra
+variables [generalized_boolean_algebra α] {s t : set α} {a b c : α}
```

Comment:
Everything in this section works for just `[semilattice_inf α] [order_bot α]`. Why do you want the extra specificity of a generalized boolean algebra?